Use fieldset `id` instead of the fieldset `title` in Form when rendering the Field component (#5921)
plone/volto: docs/source/upgrade-guide/index.md

Although it is technically possible to keep the old version running, the script `volto-update-deps` will try to update to Storybook 8 every time you run it.
```

### Form component passes down `id` of the current fieldset

> A bug where a fieldset's `title` contains spaces would cause the generated `id` to be invalid.

Reviewer: This is still confusing. See comment below. I also took another stab at it, but I am confused, so... `¯\_(ツ)_/¯`.

```suggestion
There was a bug where a fieldset's generated value would be not valid.
```

> This has been fixed by passing down the fieldset `id` instead of the `title`.

Reviewer: ```suggestion
This has been fixed by passing down the `id` instead of the `title` to the fieldset's value.
```

> If your tests rely on the old fieldset's generated `id` attribute for selecting fields, your tests could break, in which case you should amend them to use the updated fieldset's `id` instead.

Reviewer: ```suggestion
If your tests rely on the old fieldset's generated value for selecting fields, your tests could break, in which case you should amend them to use the updated fieldset's value instead.
```

Although it is technically possible to keep the old version running, the script `volto-update-deps` will try to update to Storybook 8 every time you run it.
```

> ### Form component passes down the id of the current fieldset

Reviewer: ```suggestion
### Form component passes down `id` of the current fieldset
```

> There was a bug that was provoking to have `id` attributes with spaces on it, if the a fieldset was longer than one word.

Reviewer: It's not clear to me what the symptom was, so how can I identify it?

```suggestion
A bug where a fieldset's `title` contains spaces would cause [INSERT DESCRIPTION OF SYMPTOM HERE] has been fixed.
```

Author: @stevepiercy in order to fix this one, I ended up by retouching the whole thing again.
Please take a look again.

This was because the fieldset `title` was passed down.
The fieldset `id` is passed now instead of the `title`.

> If you are relying on this for selecting fields by `id`, could be that tests could break.

Reviewer: ```suggestion
If you rely on the fieldset's `title` attribute for selecting fields, your tests could break, in which case you should amend them to use the fieldset's `id` instead.
```

> If that's the case you should amend them to use the `id`.

Reviewer: ```suggestion
```
